prometheus/promhttp/instrument_server.go

```diff
@@ -348,15 +394,27 @@ func sanitizeMethod(m string) string {
 		return "options"
 	case "NOTIFY", "notify":
 		return "notify"
+	case "TRACE", "trace":
+		return "trace"
+	case "PATCH", "patch":
+		return "patch"
 	default:
-		return strings.ToLower(m)
+		if len(additionalMethods) > 0 {
+			for _, method := range additionalMethods {
+				if strings.EqualFold(m, method) {
+					return strings.ToLower(m)
```

Comment:
Can't we just use `method` here instead of `m`? It wouldn't require `ToLower` if we documented that additional methods are used with the same capitalization as provided (which some users might even need).

Reply:
Of course, it's an option. I just respected the previous design decisions here. Everything was the lower case before and just kept it. I'm inclined to keep the lower case behaviour and if someone requests it we can change it; it's a minor one anyways.